README.md
##INSTALL
Simply run install.sh and what you want to install
e.g ./install.sh vim
./install all will copy all of my config files to your home

WARNING your config files will be overwritten!




mc/history

[user-fmt-input]
0=half type name | size | perm

[mini-input]
0=half type name | size | perm

[:select_cmd: Select ]
0=*

[inp:fishlink_cmd: Shell link to machine ]
0=ububox

[input-def]
0=*

[input2]
0=/home/master/Downloads/
1=/home/master/SHARE/Home_share/BigStorageArea/Series/
2=/home/master/SHARE/Torrent_share/71-Into.The.Fire.2010.DVDRip.XviD.AC3-ViSiON/
3=/home/master/SHARE/Torrent_share/71-Into.The.Fire.2010.DVDRip.XviD.AC3-ViSiON/71-Into.The.Fire.2010.DVDRip.XviD.AC3-ViSiON.srt
4=/home/master/SHARE/Home_share/BigStorageArea/Movies/Lucky.Luke.2009.DVDRip.XviD.AC3-ViSiON(www.USABIT.com)/Lucky.Luke.2009.DVDRip.XviD.AC3-ViSiON.srt
5=/home/master/SHARE/BigStorageArea/Master_Recordings
6=/home/master/SHARE/BigStorageArea/Master_Recordings/
7=/#sh:ububox/home/ioanna/BigStorageArea/Movies/

[cmdline]
0=r
1=ls
2=[C
3=

[Dir Hist New Left Panel]
0=/home/master/bin
1=/home/master/.bluej
2=/home/master/SHARE/Torrent_share/The Vampire Diaries season 2
3=/home/master/SHARE/Torrent_share
4=/home/master/Downloads
5=/home/master/SHARE/Home_share/Torrents
6=/home
7=/home/master/SHARE
8=/home/master/SHARE/Home_share
9=/home/master/SHARE/Home_share/BigStorageArea
10=/home/master/SHARE/Home_share/BigStorageArea/Movies
11=/home/master/SHARE/Home_share/BigStorageArea/Movies/Lucky.Luke.2009.DVDRip.XviD.AC3-ViSiON(www.USABIT.com)
12=/home/master/Music
13=/home/master/.mc
14=/home/master

[Dir Hist New Right Panel]
0=/home/master/bin
1=/home/master/SHARE/Torrent_share/The Vampire Diaries season 2
2=/home/master/SHARE/Torrent_share/Pioneer.One.S01E03.720p.x264-VODO
3=/home/master/Dropbox/cathicaki
4=/home/master/Documents/Repos/digilab
5=/home/master/Documents/Repos/digilab/askhseis
6=/home/master/Documents/Repos/digilab/askhseis/ask3
7=/home
8=/home/master/SHARE/Home_share/BigStorageArea/Series/Arrested Development
9=/home/master/Dropbox/sxolh/Ροή Λ
10=/home/master/Dropbox/sxolh/Ροή Υ/digilab/ask3
11=/home/master/SHARE/Home_share/BigStorageArea/Movies/Knight and Day (2010) R5 XviD-MAXSPEED
12=/home/master/SHARE/Home_share/BigStorageArea/Movies/Battle Los Angeles 2011 R5 XViD - IMAGiNE
13=/home/master/SHARE/Torrent_share/Priest 2011 Cam x264 Feel-Free
14=/home/master/SHARE/Torrent_share/Chatroom.2010.R5.XviD-LAP
15=/home/master/SHARE/Torrent_share/We Are The Night [Wir Sind Die Nacht].2010.BRRip.XviD-VLiS
16=/home/master/SHARE/Torrent_share/Pirates of the Caribbean On Stranger Tides TS XViD - IMAGiNE
17=/home/master/Dropbox/Pixelpipe
18=/home/master/SHARE/Torrent_share/City Under Siege 2010 dvd=-sailo1
19=/home/master/Documents
20=/home/master/Documents/Repos
21=/home/master/Documents/Repos/pewpew
22=/home/master/Documents/Repos/pewpew/pewpew/test
23=/home/master/Documents/Repos/pewpew/pewpew
24=/home/master/SHARE/Home_share/BigStorageArea/Series
25=/home/master/SHARE/Home_share/BigStorageArea/Movies/Lucky.Luke.2009.DVDRip.XviD.AC3-ViSiON(www.USABIT.com)/Subs
26=/home/master/SHARE/Home_share/BigStorageArea/Movies/Lucky.Luke.2009.DVDRip.XviD.AC3-ViSiON(www.USABIT.com)
27=/home/master/SHARE/Home_share
28=/home/master/SHARE/Home_share/BigStorageArea
29=/home/master/SHARE/Home_share/BigStorageArea/Movies/Postal 2
30=/home/master/SHARE/Home_share/BigStorageArea/Movies
31=/home/master/SHARE/Home_share/BigStorageArea/Movies/The Next Three Days (2010) DVDRip XviD-MAXSPEED
32=/home/master/Dropbox
33=/home/master/Dropbox/sxolh
34=/home/master/Dropbox/sxolh/Ροή Υ
35=/home/master/Dropbox/sxolh/Ροή Υ/digilab
36=/home/master/Dropbox/sxolh/Ροή Υ/digilab/ask4
37=/home/master/SHARE/Torrent_share/rrh
38=/home/master/SHARE/Torrent_share/Kung.Fu.Panda.2.2011.TS.AC3.XViD-EP1C
39=/home/master/SHARE/Torrent_share
40=/home/master/SHARE/Torrent_share/The Adjustment Bureau 2011 PPVRIP IFLIX www.IWANNADOWNLOAD.com
41=/home/master/SHARE
42=/home/master/SHARE/BigStorageArea
43=/home/master/SHARE/BigStorageArea/Master_Recordings
44=/home/master/.mc
45=/home/master/Downloads/Torrents
46=/home/master/Downloads
47=/home/master
48=/home/master/Repos/edaproject/demos/demo_mul_64/parallel
49=/home/master/Repos/edaproject/demos/demo_mul_64/parallel/timelogs


conky/conky/zen-rings1.lua
--[[
Ring Meters by londonali1010 (2009) mod by arpinux

This script draws percentage meters as rings. It is fully customisable; all options are described in the script.

IMPORTANT: if you are using the 'cpu' function, it will cause a segmentation fault if it tries to draw a ring straight away. The if statement on line 129 uses a delay to make sure that this doesn't happen. It calculates the length of the delay by the number of updates since Conky started. Generally, a value of 5s is long enough, so if you update Conky every 1s, use update_num>5 in that if statement (the default). If you only update Conky every 2s, you should change it to update_num>3; conversely if you update Conky every 0.5s, you should use update_num>10. ALSO, if you change your Conky, is it best to use "killall conky; conky" to update it, otherwise the update_num will not be reset and you will get an error.

To call this script in Conky, use the following (assuming that you save this script to ~/scripts/rings.lua):
	lua_load ~/scripts/rings.lua
	lua_draw_hook_pre ring_stats
]]

settings_table = {
	{
		name='time',
		arg='%I',
		max=12,
		bg_colour=0xffffff,
		bg_alpha=0,
        --		fg_colour=0x74C2B7,
        fg_colour=0x5054F8,
		fg_alpha=0.8,
		x=113, y=92,
		radius=25,
		thickness=15,
		angle=0
	},
	{
		name='time',
		arg='%M',
		max=60,
		bg_colour=0x000000,
		bg_alpha=0.1,
		fg_colour=0xffffff,
		fg_alpha=0.6,
		x=113, y=92,
		radius=40,
		thickness=10,
		angle=0
	},
	{
		name='time',
		arg='%S',
		max=60,
		bg_colour=0x000000,
		bg_alpha=0.2,
		fg_colour=0x4D4D4D,
		fg_alpha=0.8,
		x=249.5, y=170,
		radius=36,
		thickness=8,
		angle=35
	},
	{
		name='swapperc',
		arg='',
		max=100,
		bg_colour=0x000000,
		bg_alpha=0.1,
		fg_colour=0x000000,
		fg_alpha=0.6,
		x=63, y=255,
		radius=54,
		thickness=3,
		angle=51
	},
	{
		name='fs_used_perc',
		arg='/',
		max=100,
		bg_colour=0xffffff,
		bg_alpha=0,
		fg_colour=0xffffff,
		fg_alpha=0.2,
		x=63, y=255,
		radius=24,
		thickness=16,
		angle=0
	},
	{
		name='fs_used_perc',
		arg='/home',
		max=100,
		bg_colour=0xffffff,
		bg_alpha=0,
		fg_colour=0xffffff,
		fg_alpha=0.4,
		x=63, y=255,
		radius=42,
		thickness=13,
		angle=0
	},
}

require 'cairo'

function rgb_to_r_g_b(colour,alpha)
	return ((colour / 0x10000) % 0x100) / 255., ((colour / 0x100) % 0x100) / 255., (colour % 0x100) / 255., alpha
end

function draw_ring(t, pt)
	if conky_window==nil then return end
	local w,h=conky_window.width,conky_window.height
	local cs=cairo_xlib_surface_create(conky_window.display,conky_window.drawable,conky_window.visual,w,h)
	
	cr=cairo_create(cs)
	
	local xc,yc,ring_r,ring_w,angle=pt['x'],pt['y'],pt['radius'],pt['thickness'],pt['angle']
	local bgc, bga, fgc, fga=pt['bg_colour'], pt['bg_alpha'], pt['fg_colour'], pt['fg_alpha']

	local angle_0=angle*(2*math.pi/360)-math.pi/2
	local t_arc=t*2*math.pi

	-- Draw background ring

	cairo_arc(cr,xc,yc,ring_r,0,2*math.pi)
	cairo_set_source_rgba(cr,rgb_to_r_g_b(bgc,bga))
	cairo_set_line_width(cr,ring_w)
	cairo_stroke(cr)
	
	-- Draw indicator ring

	cairo_arc(cr,xc,yc,ring_r,angle_0,angle_0+t_arc)
	cairo_set_source_rgba(cr,rgb_to_r_g_b(fgc,fga))
	cairo_stroke(cr)		
	
	cairo_destroy(cr)
	cr = nil
end

function conky_cairo_cleanup()
	cairo_surface_destroy(cs)
	cs = nil
end

function conky_ring_stats()
	local function setup_rings(pt)
		local str=''
		local value=0
		
		str=string.format('${%s %s}',pt['name'],pt['arg'])
		str=conky_parse(str)
		
		value=tonumber(str)
		pct=value/pt['max']
		
		draw_ring(pct,pt)
	end
	
	-- Check that Conky has been running for at least 5s
	
	local updates=conky_parse('${updates}')
	update_num=tonumber(updates)
	
	if update_num>5 then
		for i in pairs(settings_table) do
			setup_rings(settings_table[i])
		end
	end
end
